src/hashmap.c

```diff
@@ -1,51 +1,36 @@
-#define BUCKETS 1000000
-#define PRIME_SEED_1 17
-#define PRIME_SEED_2 97
-#define KEY_CHARS 128
-#define VALUE_CHARS 128
-#define NODE_SIZE sizeof(Node)
-
+#include "hashmap.h"
 #include <stdlib.h>
-#include <stdio.h>
 #include <stdbool.h>
 #include <string.h>
-#include <assert.h>
-
-/* A key-value node in a single-linked list. */
-typedef struct Node Node;
-
-struct Node {
-    Node* next;
-    char key[KEY_CHARS];
-    char value[VALUE_CHARS]; 
-};
```

Comment:
I'm not sure how to write a nice API for a data structure in a non-OO language, but you definitely don't want to refer to a type in the header file which is only mentioned in the .c file. Makes looking at the header file on its own very confusing